File: src/test/java/ExampleSpecs.java
import com.kylecorry.frc.vision.TargetSpecs;
import org.opencv.core.Range;

/**
 * Created by Kylec on 2/12/2017.
 */
public class ExampleSpecs implements TargetSpecs {
    @Override
    public Range getHue() {
        return null;
    }

    @Override
    public Range getSaturation() {
        return null;
    }

    @Override
    public Range getValue() {
        return null;
    }

    @Override
    public double getWidth() {
        return 0;
    }

    @Override
    public double getHeight() {
        return 0;
    }

    @Override
    public double getArea() {
        return 0;
    }

    @Override
    public int getMinPixelArea() {
        return 0;
    }
}

File: src/test/java/ExampleSpecs.kt
import com.kylecorry.frc.vision.detection.TargetSpecs
import org.opencv.core.Range

/**
 * Created by Kylec on 2/12/2017.
 */
class ExampleSpecs : TargetSpecs {
    override val hue: Range
        get() = Range(0, 255)

    override val saturation: Range
        get() = Range(0, 255)

    override val value: Range
        get() = Range(0, 255)

    override val width: Double
        get() = 0.0

    override val height: Double
        get() = 0.0

    override val area: Double
        get() = 0.0

    override val minPixelArea: Int
        get() = 0
}
